docs/Configuration/fleet-server-configuration.md

```diff
@@ -213,6 +239,8 @@ By default, this will set up a Redis pool for that configuration and execute a
 
 For the address of the Redis server that Fleet should connect to, include the hostname and port.
 
+If an AWS ElastiCache endpoint is specified and `redis_password` isn't specified, Identity and Access Management (IAM) authentication is automatically used.
```

Comment:
@titanous just checking, is this a breaking change for AWS users that don't set `redis_password` today?

Put differently, will this force these users to add a role in AWS or make another change when they upgrade Fleet?

Reply:
Ah, interesting. I think it may be possible that there are Elasticache clusters without a password enabled. I don't think it's possible to configure RDS with no password but I'm not 100% certain.

Reply:
I think we can mitigate this on the Redis side by checking if `redis_username` is populated with no `redis_password`, because that config shouldn't exist in the wild and IAM auth requires a username.

Reply:
> I think it may be possible that there are Elasticache clusters without a password enabled

There are! My understanding is that Fleet is doing this for managed-cloud offerings. @rfairburn @BCTBB please correct me if I'm wrong.

Assuming this is true, I don't think we want to force a password (breaking change).

Instead, I think we want to update this behavior: IAM is only used if `redis_sts_assume_role_arn` is specified.

Meaning, If an AWS ElastiCache endpoint is specified and `redis_password` isn't specified, Fleet functions as it does today. Identity and Access Management (IAM) authentication isn't automatically used.

What do y'all think?

Reply:
We can't key off of `redis_sts_assume_role_arn`, because role assumption is not required (eg. instance/container/pod roles that are ambiently in the environment do not require additional role assumption). I think just adding a check to make sure there is a username specified should handle this case.

Reply:
Yeah, makes sense. I was trying to make the UX nice with a happy path but not a big deal to just have a flag.

Reply:
How does it work today for all other IAM Fleet server config options? ( [firehose](https://fleetdm.com/docs/configuration/fleet-server-configuration#firehose), kinesis, lambda, etc.)

It's IAM (if IAM is set), and if IAM isn't set fall back to basic auth. Right? We don't have a use IAM auth. I think we want to be consistent for now. I updated the docs to clarify this.

For example:

<img width="595" height="196" alt="Screenshot 2025-08-13 at 5 30 34 PM" src="https://github.com/user-attachments/assets/3f341b31-66d4-45bc-9096-6cea75088029" />